src/__tests__/CompareResults/ResultsView.test.tsx

```diff
@@ -150,4 +150,53 @@ describe('Results View', () => {
     expect(createObjectURLMock).toHaveBeenCalled();
     expect(revokeObjectURLMock).toHaveBeenCalledWith('blob:');
   });
+
+  it('click on retrigger button', async () => {
+    const user = userEvent.setup({ delay: null });
+
+    const { testCompareDataWithMultipleRuns, testData } = getTestData();
+    (window.fetch as FetchMockSandbox)
+      .get(
+        'begin:https://treeherder.mozilla.org/api/perfcompare/results/',
+        testCompareDataWithMultipleRuns,
+      )
+      .get('begin:https://treeherder.mozilla.org/api/project/', {
+        results: [testData[0]],
+      });
+
+    const mockGetRandomValues = jest.fn().mockReturnValue(new Uint32Array(1));
+    Object.defineProperty(window, 'crypto', {
+      value: { getRandomValues: mockGetRandomValues },
+    });
+    window.open = jest.fn();
+    jest.spyOn(Storage.prototype, 'setItem');
```

Comment:
You don't need to spy on this, you can just do things such as:
```js
expect(sessionStorage.taskclusterUrl).toBe("https://..");
expect(sessionStorage.requestState).toBe(fakeUuid);
```

Reply:
I believe we don't even need to test exactly the content of `sessionStorage.requestState`, however we need to check that the URL passed to `window.open` contains it.
So maybe something like this:
```js
const windowOpenUrlString = window.open.mock.lastCall[0];
expect(windowOpenUrlString).toBe(...); // Like you did, I believe it's still good to check the full URL.

const windowOpenUrl = new URL(windowOpenUrlString);
expect(sessionStorage.requestState).toBe(windowOpenUrl.searchParams.get("state"));
expect(sessionStorage.taskclusterUrl).toBe(windowOpenUrl.origin);
```

Also because we don't mind about the exact value of the nonce, this allows us to not mock `randomUUID` and just use whatever data node crypto library gives us.